### DISCUSSION TITLE
Is it possible to translate Fabric itself into any of language?

### DISCUSSION THREAD

**Discussion by Amirhan-Taipovjan-Greatest-I** [FabricMC/fabric-api#2858] [FAQ] [answered]
Hello!
I want to know possiblity of Fabric translation. If it's possible, then... please, explain.... how to do that?

**Comment by haykam821** [comment, marked as answer]
You can contribute translations to [Fabric API](https://github.com/FabricMC/fabric) and [Fabric installer](https://github.com/FabricMC/fabric-installer) by submitting a pull request.

For Fabric API, you must copy the translations for each applicable module into another file. There are multiple modules to translate, so you can list them all [using a search](https://github.com/search?q=repo%3AFabricMC%2Ffabric+path%3Aen_us.json+-path%3Atestmod&type=code). In particular, you can translate the tooltip shown on the item group pagination widget and the name of the resource pack generated for Fabric mods.

For Fabric installer, you must copy `installer.properties` from [here](https://github.com/FabricMC/fabric-installer/tree/02c87570135e631fa21bfe29ce95749d15717afb/src/main/resources/lang) into a new file suffixed with the language code.

**Reply by Amirhan-Taipovjan-Greatest-I** [reply]
> You can contribute translations to [Fabric API](https://github.com/FabricMC/fabric) and [Fabric installer](https://github.com/FabricMC/fabric-installer) by submitting a pull request.

Yeah, I almost translated Fabric Installer for submitting a pull request... and I translated all language files in Fabric API!


> For Fabric API, you must copy the translations for each applicable module into another file. There are multiple modules to translate, so you can list them all [using a search](https://github.com/search?q=repo%3AFabricMC%2Ffabric+path%3Aen_us.json+-path%3Atestmod&type=code). In particular, you can translate the tooltip shown on the item group pagination widget and the name of the resource pack generated for Fabric mods.

Sorry, but Your link doesn't work for Me properly... it shows that there's no any results... but I found all language files by Myself. Anyways, thank You!

> For Fabric installer, you must copy `installer.properties` from [here](https://github.com/FabricMC/fabric-installer/tree/02c87570135e631fa21bfe29ce95749d15717afb/src/main/resources/lang) into a new file suffixed with the language code.

Yeah... as I said above, translation of the Fabric Installer will be ready very soon! So, again, thank You very much!


P. S. How about translating Fabric Loader itself... is it possible to do that?
